lms/djangoapps/teams/static/teams/js/views/team_card.js

```diff
@@ -138,11 +138,17 @@
                 this.model.on('change:membership', function() {
                     this.detailViews[0].memberships = this.model.get('membership');
                 }, this);
+
+                this.teamsetName = null;
+                this.getTopic(this.model.get('topic_id')).done(_.bind(function(teamset) {
+                    this.teamsetName = teamset.get('name');
```

Comment:
one small thought: if this doesn't resolve before render() is called, the teamset name won't be displayed, and I don't know if the card will re-render if this field is updated.

don't know if that's actually a concern, and i think this is good enough for now though